### PR TITLE
feat(inter-protocol): add amm liquidity script

### PR DESCRIPTION
**Pull request by samsiegart**
refs:  https://github.com/Agoric/agoric-sdk/issues/5354

### CODE REVIEW

File: packages/inter-protocol/scripts/add-initial-liquidity.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+const dec6 = 10n ** 6n;
+const dec4 = 10n ** 4n;
+
+const config = {
+  Central: { IST: 10_000n * dec6 },
+  Secondary: { IbcATOM: 500n * dec4 }, // 20 IST per ATOM
+};
+
+// /** @type {<K extends string,V>(es: [K,V][]) => Record<K,V>} */
+// const recordFromEntries = Object.fromEntries;
```
Comment (turadg) [inline packages/inter-protocol/scripts/add-initial-liquidity.js]: remove dead code

File: packages/inter-protocol/scripts/add-initial-liquidity.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+const getOfferResult = async (instance, wallet, walletAdmin) => {
```
Comment (turadg) [inline packages/inter-protocol/scripts/add-initial-liquidity.js:16]: consider adding types

File: packages/inter-protocol/scripts/add-initial-liquidity.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  const offerIt = E(walletAdmin).getOffersNotifier();
+  let result;
+  console.log('awaiting offers...');
+  for await (const offers of makeNotifierFromAsyncIterable(offerIt)) {
+    console.log('offers iterator:', offers);
```
Comment (turadg) [inline packages/inter-protocol/scripts/add-initial-liquidity.js]: this is a next of the iterator, not the iterator itself

File: packages/inter-protocol/scripts/add-initial-liquidity.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    offers
+      .filter(
+        ({ status, invitationQuery }) =>
+          status === 'complete' && invitationQuery.instance === instance,
+      )
+      .forEach(async ({ id }) => {
+        result = await E(wallet).lookup('offerResult', id);
```
Comment (turadg) [inline packages/inter-protocol/scripts/add-initial-liquidity.js:28]: should this really run for each offer? seems like you want to `.find` the first match. In which case you don't need a separate `filter`.
```js
result = offers.find(… )
if (result) break;
```

File: packages/inter-protocol/scripts/add-initial-liquidity.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        console.log({ facet: result });
+      });
+    if (result) break;
+  }
+  return result;
+};
+
+/**
+ *
+ * @param { any } homeP
```
Comment (turadg) [inline packages/inter-protocol/scripts/add-initial-liquidity.js:38]: not blocking, but do we have a type for this somewhere?

File: packages/inter-protocol/scripts/add-initial-liquidity.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+ * @typedef {ReturnType<typeof import('../../wallet/api/src/lib-wallet').makeWalletRoot>['admin']} Wallet
+ */
+const addInitialLiquidity = async (homeP, { now = () => Date.now() }) => {
+  const { wallet, agoricNames, zoe, scratch } = E.get(homeP);
+
+  const walletBridge = E(wallet).getBridge();
+  //   const iNot = E(wb).getIssuersNotifier();
+  //   const { value: iEntries } = await E(iNot).getUpdateSince();
+  //   const issuers = recordFromEntries(iEntries);
+  //   console.log(issuers);
```
Comment (turadg) [inline packages/inter-protocol/scripts/add-initial-liquidity.js]: dead code. if it's meant to be used for additional diagnostics, put it behind a flag to enable such diagnostics

File: packages/inter-protocol/scripts/add-initial-liquidity.js [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    console.log({ proposalTemplate });
+
+    console.log('amm.makeAddLiquidityInvitation...');
+    const invitation = await E(amm.pub).makeAddLiquidityInvitation();
+    console.log({ invitation });
+
+    const offerConfig = {
+      id: `${now()}`,
+      invitation,
+      // installationHandle: amm.installation,
```
Comment (turadg) [inline packages/inter-protocol/scripts/add-initial-liquidity.js]: dead

**Review by turadg — APPROVED** [review]
I noted some non-blocking requests. I this were production code I'd Request Changes but since this is just utility a script and it's in a fast-moving part of the codebase, I'll approve and leave updates to your discretion.

File: packages/inter-protocol/scripts/add-initial-liquidity.js (hunk 5 above)
Comment (samsiegart) [reply packages/inter-protocol/scripts/add-initial-liquidity.js:38]: It used to be `import('./user-bundle').HomeBundle` which was unresolved, not sure where it came from. And I couldn't find anything else in `agoric-sdk` that defines a proper type for homeP

File: packages/inter-protocol/scripts/add-initial-liquidity.js (hunk 1 above)
Comment (samsiegart) [reply packages/inter-protocol/scripts/add-initial-liquidity.js]: removed

File: packages/inter-protocol/scripts/add-initial-liquidity.js (hunk 3 above)
Comment (samsiegart) [reply packages/inter-protocol/scripts/add-initial-liquidity.js]: fixed

File: packages/inter-protocol/scripts/add-initial-liquidity.js (hunk 6 above)
Comment (samsiegart) [reply packages/inter-protocol/scripts/add-initial-liquidity.js]: removed

File: packages/inter-protocol/scripts/add-initial-liquidity.js (hunk 7 above)
Comment (samsiegart) [reply packages/inter-protocol/scripts/add-initial-liquidity.js]: removed

**Comment by samsiegart** [comment]
> I noted some non-blocking requests. I this were production code I'd Request Changes but since this is just utility a script and it's in a fast-moving part of the codebase, I'll approve and leave updates to your discretion.

Thanks, yea I'd rather just move forward with the next PR, it takes significant time to start a local chain and test this script from scratch.

**Review by turadg — APPROVED** [review]

**Comment by dckc** [comment]
@arirubinstein note this perhaps obsoletes https://github.com/Agoric/inter-protocol-bootstrap/blob/main/add-initial-liquidity.js

**Pull request merged**